### PR TITLE
Customized config and Encryption/Decryption support

### PR DESCRIPTION
**Pull request by naveena-maplelabs**
1) Option to provide different config file rather than config.ini is added.
2) While importing/exporting user can provide new config file using argument "--config" or "-c".
python export_cluster_config -c=/path/to/ini/file
3) New file is added to encrypt and decrypt the configuration file to improve security.
python crypt.py encrypt/decrypt -k key -c /path/to/config/ini/file 4) Updated the README file.
5) Added new requirement.txt file with list of dependencies.

### CODE REVIEW

File: samples/project-anfield/README.md [hunk 1]
```diff
@@ -162,6 +162,30 @@ This will work for Python 3 >=3.4.
...
 
+## Encrypt
+
+Run the following command to encrypt the default configuration file(config.ini).
+```
+python crypt.py encrypt -k key_to_encrypt
+```
+Run the following command to encrypt the custom configuration file provided.
+```
+python crypt.py encrypt -c /path/to/config/file -k key_to_encrypt
```
Comment (pyashish) [inline samples/project-anfield/README.md:173]: python secure_config_file.py -operation encrypt -c /path/to/config.ini -key key_to_encrypt

File: samples/project-anfield/README.md [hunk 2]
```diff
@@ -162,6 +162,30 @@ This will work for Python 3 >=3.4.
...
+```
+
+## Decrypt
+
+Run the following command to decrypt the default configuration file(config.ini).
+```
+python crypt.py decrypt -k key_to_decrypt
```
Comment (pyashish) [inline samples/project-anfield/README.md:180]: python secure_config_file.py -operation decrypt -c /path/to/config.ini -key key_to_decrypt

File: samples/project-anfield/README.md [hunk 3]
```diff
@@ -170,12 +194,30 @@ python export_cluster_config.py
...
+
 Custom export file name and location can be provided as follows,
 ```
 python export_cluster_config.py --file_location /tmp/ --file_name sample_export_config
 ```
 The above command will generate sample_export_config file under /tmp folder.
 
+By default scripts read config available in config.ini file. Run the following command to export/import resources with different input config file path.
+```
+python export_cluster_config.py --config ~/backup.ini
```
Comment (pyashish) [inline samples/project-anfield/README.md:212]: python export_cluster_config.py --config clusterA_config.ini

python export_cluster_config.py --config clusterB_config.ini

File: samples/project-anfield/crypt.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
+"""
+Module to decrypt/encrypt the config file.
+How to Run: python crypt.py encrypt -c /path/to/config.ini -k key_to_encrypt/decrypt
+"""
+
+import argparse
+import sys
+import base64
+import uuid
+from cryptography.fernet import Fernet, InvalidToken
```
Comment (pyashish) [inline samples/project-anfield/crypt.py:10]: sort imports

File: samples/project-anfield/crypt.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    return key
+    """
+    # convert to bytes.
+    password = bytes(password, "utf-8")
+    salt = (uuid.getnode()).to_bytes(8, byteorder="big")
+    kdf = PBKDF2HMAC(
+        algorithm=hashes.SHA256(),
+        length=32,
+        salt=salt,
+        iterations=390000,
```
Comment (pyashish) [inline samples/project-anfield/crypt.py:29]: can you make these constants ?

File: samples/project-anfield/crypt.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+            "file.")
+        sys.exit()
+
+
+def encrypt(fernet, configfile):
+    """
+    Encrypt the file content.
+    """
+    config = read_file_content(configfile)
+    # if "export_cluster_config" not in str(config):
```
Comment (pyashish) [inline samples/project-anfield/crypt.py:71]: remove commented code

File: samples/project-anfield/crypt.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    """
+    try:
+        config = read_file_content(configfile)
+        if "export_cluster_config" in str(config):
+            print("File is already decrypted, skipping!!")
+            sys.exit()
+        decrypted_config = fernet.decrypt(config)
+        write_file_content(configfile, decrypted_config)
+    except InvalidToken:
+        print("Please provide valid key to 'decrypt' the file contents.")
```
Comment (pyashish) [inline samples/project-anfield/crypt.py:91]: we have a way to check the decryption was successful in here ?

File: samples/project-anfield/crypt.py [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        sys.exit()
+    print(f"Successfully decrypted the config file {configfile}")
```
Comment (pyashish) [inline samples/project-anfield/crypt.py:93]: after encrypting/decrypting we overwrite the files ? is there a better way ?

File: samples/project-anfield/requirement.txt [hunk 9]
```diff
@@ -0,0 +1,2 @@
+cohesity_management_sdk==1.8.1
+cryptography==2.1.4
```
Comment (pyashish) [inline samples/project-anfield/requirement.txt:2]: Is there a conditional requirement? cryptography is a requirement only if they want to do encryption correct ?

**Review by pyashish — CHANGES_REQUESTED** [review]

File: samples/project-anfield/requirement.txt (hunk 9 above)
Comment (naveena-maplelabs) [reply samples/project-anfield/requirement.txt:2]: Commented and added a note in the file

File: samples/project-anfield/crypt.py (hunk 5 above)
Comment (naveena-maplelabs) [reply samples/project-anfield/crypt.py:29]: added

**Pull request merged**